core/src/test/java/io/grpc/internal/ServiceConfigErrorHandlingTest.java

```diff
@@ -0,0 +1,655 @@
+    ImmutableMap<String, Object> ignoredServiceConfig =
+        ImmutableMap.<String, Object>of("loadBalancingPolicy", "round_robin");
+    nameResolverFactory.servers.clear();
+    nameResolverFactory.nextConfigOrError.set(ConfigOrError.fromConfig(ignoredServiceConfig));
+
+    // 2nd resolution
+    nameResolverFactory.allResolved();
+
+    // 2nd service config without address should be ignored
+    verify(mockLoadBalancer, never()).handleResolvedAddresses(any(ResolvedAddresses.class));
+    verify(mockLoadBalancer, never()).handleNameResolutionError(any(Status.class));
+    assertThat(channel.getState(false)).isEqualTo(ConnectivityState.IDLE);
+  }
+
+  @Test
+  public void emptyAddresses_validConfig_lbDoesNotNeedAddress() throws Exception {
+    FakeNameResolverFactory nameResolverFactory =
+        new FakeNameResolverFactory.Builder(expectedUri)
+            .setServers(Collections.<EquivalentAddressGroup>emptyList())
+            .build();
+    channelBuilder.nameResolverFactory(nameResolverFactory);
+    when(mockLoadBalancer.canHandleEmptyAddressListFromNameResolution()).thenReturn(true);
+
+    ImmutableMap<String, Object> serviceConfig =
+        ImmutableMap.<String, Object>of("loadBalancingPolicy", "mock_lb");
+    nameResolverFactory.nextConfigOrError.set(ConfigOrError.fromConfig(serviceConfig));
+
+    createChannel();
+
+    ArgumentCaptor<ResolvedAddresses> resultCaptor =
+        ArgumentCaptor.forClass(ResolvedAddresses.class);
+    verify(mockLoadBalancer).handleResolvedAddresses(resultCaptor.capture());
+    assertThat(resultCaptor.getValue().getAddresses()).isEmpty();
+    Attributes actualAttrs = resultCaptor.getValue().getAttributes();
+    assertThat(actualAttrs.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG))
+        .isEqualTo(serviceConfig);
+    verify(mockLoadBalancer, never()).handleNameResolutionError(any(Status.class));
+
+    assertThat(channel.getState(false)).isNotEqualTo(ConnectivityState.TRANSIENT_FAILURE);
+  }
+
+  @Test
+  public void validConfig_lbDoesNotNeedAddress() throws Exception {
+    FakeNameResolverFactory nameResolverFactory =
+        new FakeNameResolverFactory.Builder(expectedUri)
+            .setServers(ImmutableList.of(addressGroup))
+            .build();
+    channelBuilder.nameResolverFactory(nameResolverFactory);
+
+    ImmutableMap<String, Object> serviceConfig =
+        ImmutableMap.<String, Object>of("loadBalancingPolicy", "mock_lb");
+    nameResolverFactory.nextConfigOrError.set(ConfigOrError.fromConfig(serviceConfig));
+
+    createChannel();
+
+    ArgumentCaptor<ResolvedAddresses> resultCaptor =
+        ArgumentCaptor.forClass(ResolvedAddresses.class);
+    verify(mockLoadBalancer).handleResolvedAddresses(resultCaptor.capture());
+    assertThat(resultCaptor.getValue().getAddresses()).containsExactly(addressGroup);
+    Attributes actualAttrs = resultCaptor.getValue().getAttributes();
+    assertThat(actualAttrs.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG))
```

Comment:
This is not what LBs are looking for. You should check `LoadBalancer.ATTR_LOAD_BALANCING_CONFIG` instead, and the expected value should be the value of the `mock_lb` key in the service config, which is
```
          {
            "param1": "value1",
            "param2": "value2",
          }
```
from my comment above.

Reply:
added TODO. iirc we discussed in offline that we want to make this change after this PR because it requires changes in LB to reduce the scope of this pr.